synergy-design-system/synergy-design-system
# chore: 🔧 e2e testing on production builds

# Pull Request

## 📖 Description

This PR refactors our e2e test pipeline for our demo projects to make them run smoother. For this to work, the new pipeline configuration is:

- using the production builds of the demo projects (except for React, see below)
- Not using `npm-run-all`, but a static webserver
- Making sure to just run one instance of playwright for more stable tests



### 🎫 Issues

Closes #505

## 👩‍💻 Reviewer Notes

Initially, I was hoping that I would be able to include a react production build, too. However, it turned out that the production build of our react demo runs fine __outside__ of the mono-repo, but not inside of it. There seem to be issues like #490 which effect the production build for vite on the react target only. Unfortunately, I was not able to fix those issues, so React runs on vite dev in the meantime. It may have to be something to do with how rollup bundles the files, but no configuration override or package update helped to ease this problem :(.

## 📑 Test Plan

- Check out the branch and run a clean install `pnpm clean` in the root of the repo
- Move to `packages/_private/e2e-demo-test`
- Run `pnpm start` to spin up the system
- Notice how the tests now run in sequence with only one instance of Playwright
- Also notice that the tests have valid names now, which should make debugging easier.

## ✅ DoD
